test/threads.asm.js

```diff
@@ -0,0 +1,141 @@
+  var nullFunc_vi=env.nullFunc_vi;
+  var nullFunc_vii=env.nullFunc_vii;
+  var nullFunc_ii=env.nullFunc_ii;
+  var nullFunc_viii=env.nullFunc_viii;
+  var nullFunc_v=env.nullFunc_v;
+  var nullFunc_iii=env.nullFunc_iii;
+  var invoke_iiii=env.invoke_iiii;
+  var invoke_i=env.invoke_i;
+  var invoke_vi=env.invoke_vi;
+  var invoke_vii=env.invoke_vii;
+  var invoke_ii=env.invoke_ii;
+  var invoke_viii=env.invoke_viii;
+  var invoke_v=env.invoke_v;
+  var invoke_iii=env.invoke_iii;
+  var __spawn_thread=env.__spawn_thread;
+  var _putenv=env._putenv;
+  var _emscripten_get_now_is_monotonic=env._emscripten_get_now_is_monotonic;
+  var _fpathconf=env._fpathconf;
+  var ___unlock=env.___unlock;
+  var _emscripten_syscall=env._emscripten_syscall;
+  var ___assert_fail=env.___assert_fail;
+  var _utimes=env._utimes;
+  var ___buildEnvironment=env.___buildEnvironment;
+  var _emscripten_asm_const_i=env._emscripten_asm_const_i;
+  var _clock_gettime=env._clock_gettime;
+  var _emscripten_futex_wait=env._emscripten_futex_wait;
+  var _tzset=env._tzset;
+  var ___setErrNo=env.___setErrNo;
+  var _emscripten_set_current_thread_status_js=env._emscripten_set_current_thread_status_js;
+  var _pthread_getschedparam=env._pthread_getschedparam;
+  var _clearenv=env._clearenv;
+  var _emscripten_futex_wake=env._emscripten_futex_wake;
+  var _sysconf=env._sysconf;
+  var _utime=env._utime;
+  var ___call_main=env.___call_main;
+  var _emscripten_memcpy_big=env._emscripten_memcpy_big;
+  var _confstr=env._confstr;
+  var _getenv=env._getenv;
+  var ___syscall54=env.___syscall54;
+  var _emscripten_has_threading_support=env._emscripten_has_threading_support;
+  var _pthread_create=env._pthread_create;
+  var _emscripten_get_now=env._emscripten_get_now;
+  var _chroot=env._chroot;
+  var ___lock=env.___lock;
+  var ___syscall6=env.___syscall6;
+  var _unsetenv=env._unsetenv;
+  var ___clock_gettime=env.___clock_gettime;
+  var _gettimeofday=env._gettimeofday;
+  var _atexit=env._atexit;
+  var ___syscall140=env.___syscall140;
+  var _emscripten_set_thread_name_js=env._emscripten_set_thread_name_js;
+  var _setenv=env._setenv;
+  var ___syscall146=env.___syscall146;
+  var _emscripten_conditional_set_current_thread_status_js=env._emscripten_conditional_set_current_thread_status_js;
+  var Atomics_load=global.Atomics.load;
+  var Atomics_store=global.Atomics.store;
+  var Atomics_exchange=global.Atomics.exchange;
+  var Atomics_compareExchange=global.Atomics.compareExchange;
+  var Atomics_add=global.Atomics.add;
+  var Atomics_sub=global.Atomics.sub;
+  var Atomics_and=global.Atomics.and;
+  var Atomics_or=global.Atomics.or;
+  var Atomics_xor=global.Atomics.xor;
+  var tempFloat = Math_fround(0);
+  const f0 = Math_fround(0);
+
+  function test() {
+   var $temp = 0;
+   $temp = (Atomics_load(HEAP32, 1229)|0);
+   $temp = (Atomics_load(HEAPU16, 1229)|0);
+   $temp = (Atomics_store(HEAP32, $temp>>2, 0)|0);
+   $temp = (Atomics_exchange(HEAP32, $temp>>2, 1)|0);
```

Comment:
Does this change work with smaller sized Atomic RMW operations too?

Reply:
It should, yeah, I added a commit with tests for a few more smaller size examples.

Reply:
Thanks, looks good.